modules/ROOT/pages/dataweave-cookbook-decode-encode.adoc

```diff
@@ -0,0 +1,158 @@
++
+This action creates a PDF file in the specified location and generates a separate `transform` file with the message in the Set Payload component, `PDF file created in the directory specified in the Write operation.`
+
+[[ex01]]
+== Encode a PDF File to Base64
+
+You can simulate the transformation in the Read operation from the <<xml_config, XML configuration>>. This example performs Base64 encoding on a PDF file. More specifically, it converts a Base64 file stream (`application/octet-stream`) to Binary and outputs the Binary in the `text/plain` format.
+
+The example uses the following functions:
+
+*  `readUrl` from the `Core (dw::Core)` module to input a PDF file.
++
+The example assumes that the PDF file is in the `src/main/resources` directory of a Mule project in Anypoint Studio. It provides the name of the PDF file and specifies the MIME type as `application/octet-stream`.
+* `fromBase64` from the `dw::core::Binaries` module to transform a Base64 string to a binary value.
+
+//NOTE that we can't yet automate this example. Outputs text format.
+//Need to check if PDF input is supported by validator.
+.DataWeave Script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+import * from dw::core::Binaries
+var myPDF = readUrl("classpath://pdf-test.pdf", "application/octet-stream")
+output text/plain
+---
+toBase64(myPDF as Binary)
+----
+
+.Output (partial results):
+[source,json,linenums]
+----
+JVBERi0xLjYNJeLjz9MNCjM3IDAgb2JqIDw8L0xpbmVhcml6ZWQgMS9MIDIwNTk3L08gNDAvRSAx
+...
+----
+
+
+[[ex02]]
+== Decode Base64
+
+You can simulate the transformation in the Write operation from the <<xml_config, XML configuration>>. This DataWeave example converts a Base64 file stream (`application/octet-stream`) to Binary and outputs the Binary in the `text/plain` format. It does not generate a PDF file; a Write operation is required to generate such a file.
+
+The examples use the following functions from the `dw::core::Binaries` module:
+
+*  `readUrl` to input a PDF file.
++
+The example assumes that the PDF file is in the `src/main/resources` directory of a Mule project in Anypoint Studio. It provides the name of the PDF file and specifies the MIME type as `application/octet-stream`.
+* `fromBase64` in the `dw::core::Binaries` module to transform a Base64 string to a binary value.
+
+The script reads the input PDF as an octet-stream MIME type. From within the call to `fromBase64`, the script coerces that input to a string value before coercing the string to Base64.
+
+//NOTE that we can't yet automate this example. Outputs text format.
+//Need to check if PDF input is supported by validator.
+.DataWeave Script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+import * from dw::core::Binaries
+var myPDF = readUrl("classpath://pdf-test.pdf", "application/octet-stream")
+var myPDFasBinary = toBase64(myPDF as Binary)
+output text/plain
+---
+fromBase64(myPDFasBinary as String) as Binary
+----
+
+.Output (partial results):
+[source,json,linenums]
+----
+%PDF-1.6
```

Comment:
Maybe clarify this is the text representation of a pdf?

Reply:
done

Reply:
I think if you try this now it will output gibberish because it's now binary. The output needs updating and the note now has to mention it's the binary representation.